### PR TITLE
Break the reference cycle between the surface factory and the external view embedder

### PR DESCRIPTION
**Pull request by jason-simmons**
See https://github.com/flutter/flutter/issues/68315

### CODE REVIEW

File: shell/platform/android/platform_view_android.cc [hunk 1]
```diff
@@ -107,7 +107,9 @@ PlatformViewAndroid::PlatformViewAndroid(
       jni_facade_(jni_facade),
       platform_view_android_delegate_(jni_facade) {}
 
-PlatformViewAndroid::~PlatformViewAndroid() = default;
+PlatformViewAndroid::~PlatformViewAndroid() {
+  surface_factory_->SetExternalViewEmbedder(nullptr);
```
Comment (gaaclarke) [inline shell/platform/android/platform_view_android.cc:111]: surface_factory_ is a shared_ptr, you could be editing a variable that is being held by someone unexpectedly.  I don't see anywhere where the surface_factory_ is actually shared.  If it isn't shared we should move it to a unique_ptr to make this operation obviously safe.

Comment (gaaclarke) [reply shell/platform/android/platform_view_android.cc:111]: Ahh wait, the AndroidExternalViewEmbedder is holding onto it.  Do you think we should make that a weak_ptr instead?

Comment (jason-simmons) [reply shell/platform/android/platform_view_android.cc]: Moved the strong reference to the ExternalViewEmebedder up to the PlatformView.
Changed the SurfaceFactory's reference to a weak pointer.

**Review by gaaclarke — APPROVED** [review]
Here is the relationships right?
```
AndroidSurfaceFactoryImpl -weak-> AndroidExternalViewEmbedder
AndroidExternalViewEmbedder -strong-> AndroidSurfaceFactoryImpl
PlatformViewAndroid -strong-> AndroidSurfaceFactoryImpl
PlatformViewAndroid -strong-> AndroidExternalViewEmbedder
```

That looks good to me.  I think having the weak_ptr is better since the annotation is more correct and we don't have to worry about untying a knot in a deconstructor.

**Pull request merged**